src/IROperator.cpp

```diff
@@ -1048,6 +1048,18 @@ Expr predicate(Expr e) {
                                 {std::move(e)}, Internal::Call::PureIntrinsic);
 }
 
+Expr predicate_loads(Expr e) {
+    Type t = e.type();
+    return Internal::Call::make(t, Internal::Call::predicate_loads,
+                                {std::move(e)}, Internal::Call::PureIntrinsic);
```

Comment:
I don't think this new intrinsic is pure. pure intrinsics can be moved around in this IR without changing the meaning. This one must stay where it is immediately inside the if condition.

Reply:
I changed this (and the pre-existing Call::predicate), but this seems to break some logic in ScheduleFunctions, which looks for impure calls in conditions.

I changed this back to PureIntrinsic for now. I agree this is an issue, but it's pre-existing with Call::predicate too, and I'm not sure how this should change yet.